### PR TITLE
Add support for semantic version numers to listing indexes

### PR DESCRIPTION
**Pull request by cfvonner**
Puts tags/functions added in an update version into the list for the major release.

Will handle current versioning (2016.0.#) and past versioning (11.0.#).

### CODE REVIEW

File: utilities/indexer.cfm [hunk 1]
```diff
@@ -23,10 +23,16 @@
...
 				<cfset arrayAppend(categories, nameKey)>
 			</cfif>
-			<cfif structKeyExists(data, "engines") AND structKeyExists(data.engines, "coldfusion") AND structKeyExists(data.engines.coldfusion, "minimum_version") AND isNumeric(data.engines.coldfusion.minimum_version)>
-
-				<cfif structKeyExists(versions, val(data.engines.coldfusion.minimum_version))>
-					<cfset arrayAppend(versions[val(data.engines.coldfusion.minimum_version)], data.name)>
+			<cfif structKeyExists(data, "engines") AND structKeyExists(data.engines, "coldfusion") AND structKeyExists(data.engines.coldfusion, "minimum_version")>
+				<cfif isNumeric(data.engines.coldfusion.minimum_version)>
+					<cfset minimum_version = data.engines.coldfusion.minimum_version>
+				<cfelseif reFind('^[0-9]{2,4}\.[0-9]\.[0-9]+$', data.engines.coldfusion.minimum_version)>
```
Comment (shaedrich) [inline utilities/indexer.cfm]: :+1: :heart:

File: utilities/indexer.cfm [hunk 2]
```diff
@@ -23,10 +23,16 @@
...
+					<cfset minimum_version = reFind( '^([0-9]{2,4})', data.engines.coldfusion.minimum_version, 1, true ).match[1]>
+				<cfelse>
+					<cfset minimum_version = ''>
+				</cfif>
+				<cfif structKeyExists(versions, val(minimum_version))>
```
Comment (shaedrich) [inline utilities/indexer.cfm]: Not your fault but a note to old code: Railo has version 0. If you set `minimum_version` to `''` and run it through `val()`, you'll get 0 which might exist in versions struct but currently does not so there was not unexpected behavoir yet.

Comment (cfvonner) [reply utilities/indexer.cfm]: I didn't know that! 👍  

What do you think would be a better placeholder for non-compliant version numbers?  `-1` maybe?

Comment (shaedrich) [reply utilities/indexer.cfm]: Or just don't set it and then check for existence.

Comment (cfvonner) [reply utilities/indexer.cfm]: I'll work on an update to this tomorrow.  I like the idea of not setting it and checking for existence.

**Pull request merged**